File: src/stores/project.ts
import { defineStore } from 'pinia'
import { ref, computed } from 'vue'
import api from '../api'

export interface Project {
  project_id: string
  title: string
  total_lines: number
  total_chapters: number
  current_line: number
  current_chapter: number
  novel_path?: string
  project_path: string
  created_at: string
  updated_at: string
}

export const useProjectStore = defineStore('project', () => {
  // 当前选中的项目ID
  const selectedProjectId = ref<string>('')
  
  // 项目列表
  const projects = ref<Project[]>([])
  
  // 当前项目详情
  const currentProject = ref<Project | null>(null)
  
  // 计算属性：当前选中的项目
  const selectedProject = computed(() => {
    return projects.value.find(p => p.project_id === selectedProjectId.value) || null
  })
  
  // 加载项目列表
  const loadProjects = async () => {
    try {
      const data = await api.get('/project/all', {
        params: { limit: 100, offset: 0 }
      })
      projects.value = Array.isArray(data) ? data : []
      
      // 如果当前选择的项目不在列表中，清空选择
      if (selectedProjectId.value && !selectedProject.value) {
        selectedProjectId.value = ''
        currentProject.value = null
      }
      
      // 尝试恢复上次选择的项目
      if (!selectedProjectId.value && projects.value.length > 0) {
        const savedProjectId = localStorage.getItem('currentProjectId')
        if (savedProjectId && projects.value.find(p => p.project_id === savedProjectId)) {
          setSelectedProjectId(savedProjectId)
        } else {
          // 如果没有保存的项目，选中第一个
          setSelectedProjectId(projects.value[0].project_id)
        }
      }
    } catch (error) {
      console.error('加载项目列表失败:', error)
      projects.value = []
    }
  }
  
  // 设置选中的项目ID
  const setSelectedProjectId = (projectId: string) => {
    selectedProjectId.value = projectId
    if (projectId) {
      localStorage.setItem('currentProjectId', projectId)
      loadCurrentProject()
    } else {
      localStorage.removeItem('currentProjectId')
      currentProject.value = null
    }
  }
  
  // 加载当前项目详情
  const loadCurrentProject = async () => {
    if (!selectedProjectId.value) {
      currentProject.value = null
      return
    }
    
    try {
      currentProject.value = await api.get(`/project/${selectedProjectId.value}`)
    } catch (error) {
      console.error('加载项目详情失败:', error)
      currentProject.value = null
    }
  }
  
  // 初始化：从 localStorage 恢复项目ID
  const init = () => {
    const savedProjectId = localStorage.getItem('currentProjectId')
    if (savedProjectId) {
      selectedProjectId.value = savedProjectId
    }
  }
  
  return {
    selectedProjectId,
    projects,
    currentProject,
    selectedProject,
    loadProjects,
    setSelectedProjectId,
    loadCurrentProject,
    init
  }
})

File: src/api/schemas/project.py
"""
项目相关的数据模型。

每次小说转漫画任务都是一个项目，包含项目配置、进度状态等信息。
"""
import uuid
from datetime import datetime
from typing import Optional
from sqlmodel import SQLModel, Field


def generate_project_id() -> str:
    """
    生成项目ID（不带连字符的UUID）。
    
    返回32位十六进制字符串，只包含数字和字母，保持UUID的随机性。
    例如：f07916f0f4974754a6c404273e557519
    """
    return uuid.uuid4().hex


class Project(SQLModel, table=True):
    """项目主体。"""
    project_id: Optional[str] = Field(description="项目唯一标识", primary_key=True, default_factory=generate_project_id)
    title: str = Field(description="项目标题", index=True)
    novel_path: Optional[str] = Field(default=None, description="小说文件路径")
    project_path: str = Field(description="项目存储路径")

    # 小说元数据
    total_lines: int = Field(default=0, ge=0, description="小说总行数（段落数）")
    total_chapters: int = Field(default=0, ge=0, description="小说总章节数")
    current_line: int = Field(default=0, description="当前处理段落")
    current_chapter: int = Field(default=0, description="当前处理章节", index=True)

    created_at: datetime = Field(default_factory=datetime.now, description="创建时间")
    updated_at: datetime = Field(default_factory=datetime.now, description="更新时间")
